src/main/kotlin/org/rust/cargo/toolchain/wsl/Utils.kt

```diff
@@ -14,6 +18,26 @@ fun WSLDistribution.expandUserHome(path: String): String {
     return "$userHome${path.substring(1)}"
 }
 
-fun Path.hasExecutableOnWsl(toolName: String): Boolean = pathToExecutableOnWsl(toolName).toFile().isFile
+fun Path.hasExecutableOnWsl(toolName: String): Boolean {
+    checkEdtAndReadAction()
```

Comment:
Are you sure that it's proper place for this check?
If I understand correctly, it will work only on Windows. So most of the plugin developers will never see the result of this check at all. But at the same time, they may write code that will have poor performance on Windows because of WSL

Reply:
There are +2 exceptions just after opening a project, and I'm not sure how to fix them at the moment, so I think if we enable it not only for WSL, it will be annoying until I fix the existing exceptions.

Reply:
Maybe then it's better not to introduce this check at all now, fix these two cases and turn it on only after it?

Reply:
Done.